packages/ai-jsx/src/core/image-gen.tsx

```diff
@@ -77,14 +79,46 @@ export function ImageGenProvider<T extends ImageGenPropsWithChildren>(
  *    </ImageGen>
  * ```
  */
-export function ImageGen(
-  { children, ...props }: ImageGenPropsWithChildren & Record<string, unknown>,
-  { getContext }: RenderContext
-) {
+export function ImageGen({ children, ...props }: ImageGenPropsWithChildren, { getContext }: RenderContext) {
   const [ImageGenComponent, defaultProps] = getContext(imageGenContext);
   return (
     <ImageGenComponent {...defaultProps} {...props}>
       {children}
     </ImageGenComponent>
   );
 }
+
+/**
+ * A wrapper around {@link ImageGen} that renders the generated image(s) as an HTML "img" tag.
+ * To be used only in the browser, not the CLI.
+ *
+ * @returns HTML "img" tag(s) with the generated image(s).
+ */
+export async function ImageGenHTML(
+  {
+    children,
+    width = undefined,
+    height = undefined,
+    alt = undefined,
+    ...props
+  }: ImageGenPropsWithChildren & {
+    /** Width for the HTML image tag. */ width?: string;
+    /** Height for the HTML image tag. */ height?: string;
+    /** Alternative text for the HTML image tag.  */ alt?: string;
+  },
+  { render }: RenderContext
+) {
+  const urls = await render(<ImageGen {...props}>{children}</ImageGen>);
+  return (
+    <AIR.React>
```

Comment:
Per our conversation this morning I think this is backwards -- I feel strongly that users shouldn't have to select the right `ImageGenX` variant with knowledge of the eventual rendering surface. (This basically means any component that wants to do an `ImageGen` internally needs to either know where it's going to end up or invent an affordance to plumb it down.)

- My first choice is to add code at the React/AI integration boundary to handle this
- My second choice would be to use context providers to allow ancestors to override the rendering behavior

At a high level I think it's quite important to decouple the abstract representation of the image in AI.JSX (where it's not hard to imagine components that would pull the image and read the bytes) from the way it gets rendered in eventual UI.

Reply:
to be precise, https://github.com/fixie-ai/ai-jsx/pull/160 is what I'm proposing

Reply:
I feel slightly queasy about that but I also see the appeal of what you're saying, and I think if we chatted about it more, I'd update in your direction. So I'm fine to go with this approach.

Reply:
Going with Peter's approach as discussed.